server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4302,13 +4311,15 @@ protected long doGenerateSeqNoForOperation(Operation operation) {
             noOpEngine.recoverFromTranslog(translogHandler, Long.MAX_VALUE);
             final int gapsFilled = noOpEngine.fillSeqNoGaps(primaryTerm.get());
             final String reason = "filling gaps";
-            noOpEngine.noOp(new Engine.NoOp(maxSeqNo + 1, primaryTerm.get(), LOCAL_TRANSLOG_RECOVERY, System.nanoTime(), reason));
-            assertThat(noOpEngine.getLocalCheckpoint(), equalTo((long) (maxSeqNo + 1)));
+            noOpEngine.noOp(new Engine.NoOp(maxSeqNo + 1, primaryTerm.get(), LOCAL_TRANSLOG_RECOVERY,
+                System.nanoTime(), reason));
+            assertThat(noOpEngine.getProcessedLocalCheckpoint(), equalTo((long) (maxSeqNo + 1)));
             assertThat(noOpEngine.getTranslog().stats().getUncommittedOperations(), equalTo(gapsFilled));
-            noOpEngine.noOp(
+            Engine.NoOpResult result = noOpEngine.noOp(
                 new Engine.NoOp(maxSeqNo + 2, primaryTerm.get(),
                     randomFrom(PRIMARY, REPLICA, PEER_RECOVERY), System.nanoTime(), reason));
-            assertThat(noOpEngine.getLocalCheckpoint(), equalTo((long) (maxSeqNo + 2)));
+            noOpEngine.ensureTranslogSynced(Stream.of(result.getTranslogLocation()));
```

Comment:
I think this might be a leftover from previous iterations? As far as I can see, the test relies only on the processed, not the persisted checkpoints? There are a few similar cases in this class (though some might be necessary).

Reply:
right, I have gone through the class again and adapted all throughout. I've used the persisted local checkpoint whenever we simulated updating the global checkpoint, to make it more realistic.